node/nodeBlocks_test.go

```diff
@@ -0,0 +1,263 @@
+package node_test
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	apiBlock "github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/node"
+	"github.com/ElrondNetwork/elrond-go/node/mock"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetBlockByHash_InvalidShardShouldErr(t *testing.T) {
+	t.Parallel()
+
+	n, _ := node.NewNode()
+
+	blk, err := n.GetBlockByHash("invalidHash", false)
+	assert.Error(t, err)
+	assert.Nil(t, blk)
+}
+
+func TestGetBlockByHashFromHistoryNode(t *testing.T) {
+	t.Parallel()
+
+	historyProc := &mock.HistoryProcessorStub{
+		IsEnabledCalled: func() bool {
+			return true
+		},
+		GetEpochForHashCalled: func(hash []byte) (uint32, error) {
+			return 1, nil
+		},
+	}
+
+	nonce := uint64(1)
+	round := uint64(2)
+	epoch := uint32(1)
+	shardID := uint32(5)
+	miniblockHeader := []byte("mbHash")
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+	n, _ := node.NewNode(
+		node.WithInternalMarshalizer(&mock.MarshalizerFake{}, 90),
+		node.WithHistoryProcessor(historyProc),
+		node.WithShardCoordinator(mock.NewOneShardCoordinatorMock()),
+		node.WithDataStore(&mock.ChainStorerMock{
```

Comment:
Perhaps use a real (light though) storer instead? (easier to follow the test perhaps)

Reply:
now i use a storer mock instead of stub.